### PR TITLE
feat(perf): Generalize histogram function

### PR DESCRIPTION
**Pull request by Zylphrex**
The existing `measurements_histogram` function is hard coded to work only for
measurements. Given the need to also generate similar histograms for other
numeric columns, this change generalizes this function to permit other columns
such as `transaction.duration` to be used as well.

### CODE REVIEW

File: src/sentry/api/event_search.py [hunk 1]
```diff
@@ -2021,8 +2031,9 @@ def result_type_fn(function_arguments, parameter_values):
             private=True,
         ),
         Function(
-            "measurements_histogram",
+            "histogram2",
```
Comment (wmak) [inline src/sentry/api/event_search.py:2034]: Is the plan to eventually replace the existing `histogram` with this function? Should we rename the old function to `histogram_deprecated` (or similar) instead?

Comment (Zylphrex) [reply src/sentry/api/event_search.py:2034]: Yes, the intent is to eventually replace the existing `histogram` function. Would you prefer I deprecated the old function as part of this PR?

Comment (wmak) [reply src/sentry/api/event_search.py:2034]: Yeah, that way we can rename `measurements_histogram` to just `histogram` which makes it clearer that its the  right histogram function to use.

File: src/sentry/static/sentry/app/views/performance/transactionSummary/latencyChart.tsx [hunk 2]
```diff
@@ -71,8 +71,8 @@ class LatencyChart extends AsyncComponent<Props, State> {
       id: '',
       name: '',
       version: 2,
-      fields: [`histogram(transaction.duration,${NUM_BUCKETS})`, 'count()'],
-      orderby: 'histogram_transaction_duration_15',
+      fields: [`histogram_deprecated(transaction.duration,${NUM_BUCKETS})`, 'count()'],
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/performance/transactionSummary/latencyChart.tsx:74]: Getting this to use the same histogram with fencing like we use on measures would be 🤩

Comment (Zylphrex) [reply src/sentry/static/sentry/app/views/performance/transactionSummary/latencyChart.tsx:74]: That's the goal 👍 will be adding a new histogram end point that we'll be able to use to support all histograms going forwards.

**Review by wmak — APPROVED** [review]
Nice 👏

**Pull request merged**